### PR TITLE
add information about testing in workspace and components

### PR DESCRIPTION
**Pull request by doudou**

### CODE REVIEW

**Comment by doudou** [comment]
@henriquejsfj could you have a read ? I'm _very_ aware that it's probably hard to understand how to understand all of this ... I'd be very glad for a thorough review.

**Comment by doudou** [comment]
Nice. You probably will have more comments once you've actually written the tests of the websocket controldev task, let's wait until then before merging this.

File: source/components/testing.html.md [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+using_task_library 'project_name'
+
+describe OroGen.project_name.Task do
+    run_live
+end
+~~~
+
+## Deploying a component
+
+The test harness is provides discrete steps to change the component step:
```
Comment (g-arjones) [inline source/components/testing.html.md]: The test harness ~~is~~ provides discrete

**Comment by g-arjones** [comment]
Pretty comprehensive... I missed one common case though: testing that ONE output port sends TWO samples in a row. I remember I needed this once and I couldn't manage, ended up creating a socket in my tests and connecting the iodriver's base subject to it by setting task.properties.io_port

I mean something like:

```ruby
expect_execution { task.set_and_check_something }
    .to do
        [have_one_new_sample(task.io_raw_out_port),
         have_one_new_sample(task.io_raw_out_port)]
    end
```

**Comment by doudou** [comment]
In hindsight, a rather obvious need, actually. 

Would need to refactor the `have_one_new_sample`  add a more generic `have_new_samples` expectation with a count and then use it to "reimplement" have_one_new_sample.

Added an issue on syskit https://github.com/rock-core/tools-syskit/issues/232

**Comment by g-arjones** [comment]
Thanks... :+1:

**Pull request merged**